test/webaudio/audioworklet_in_out_stereo.c

```diff
@@ -0,0 +1,169 @@
+    return false;
+  }
+  return true;
+}
+
+// ID to the beat and bass loops
+EMSCRIPTEN_WEBAUDIO_T beatID = 0;
+EMSCRIPTEN_WEBAUDIO_T bassID = 0;
+
+// Creates a MediaElementAudioSourceNode with the supplied URL (which is
+// registered as an internal audio object and the ID returned).
+EM_JS(EMSCRIPTEN_WEBAUDIO_T, createTrack, (EMSCRIPTEN_WEBAUDIO_T ctxID, const char* url, bool looping), {
+  var context = emscriptenGetAudioObject(ctxID);
+  if (context) {
+    var audio = document.createElement('audio');
+    audio.src = UTF8ToString(url);
+    audio.loop = looping;
+    var track = context.createMediaElementSource(audio);
+    return emscriptenRegisterAudioObject(track);
+  }
+  return 0;
+});
+
+// Toggles the play/pause of a MediaElementAudioSourceNode given its ID
+EM_JS(void, toggleTrack, (EMSCRIPTEN_WEBAUDIO_T srcID), {
+  var source = emscriptenGetAudioObject(srcID);
+  if (source) {
+    var audio = source.mediaElement;
+    if (audio) {
+      if (audio.paused) {
+        audio.currentTime = 0;
+        audio.play();
+      } else {
+        audio.pause();
+      }
+    }
+  }
+});
+
+// Callback to process and mix the audio tracks
+bool process(int numInputs, const AudioSampleFrame* inputs, int numOutputs, AudioSampleFrame* outputs, int numParams, const AudioParamFrame* params, void* data) {
+  audioProcessedCount++;
+
+  // Single stereo output
+  assert(numOutputs == 1 && outputs[0].numberOfChannels == 2);
+  for (int n = 0; n < numInputs; n++) {
+    // And all inputs are also stereo
+    assert(inputs[n].numberOfChannels == 2 || inputs[n].numberOfChannels == 0);
+    // This should always be the case
+    assert(inputs[n].samplesPerChannel == outputs[0].samplesPerChannel);
+  }
+  // We can now do a quick mix since we know the layouts
+  if (numInputs > 0) {
+    int totalSamples = outputs[0].samplesPerChannel * outputs[0].numberOfChannels;
+    float* outputData = outputs[0].data;
+    memcpy(outputData, inputs[0].data, totalSamples * sizeof(float));
+    for (int n = 1; n < numInputs; n++) {
+      // It's possible to have an input with no channels
+      if (inputs[n].numberOfChannels == 2) {
+        float* inputData = inputs[n].data;
+        for (int i = totalSamples - 1; i >= 0; i--) {
+          outputData[i] += inputData[i];
+        }
+      }
+    }
+  }
+  return true;
+}
+
+// Registered click even to (1) enable audio playback and (2) toggle playing the tracks
+bool onClick(int type, const EmscriptenMouseEvent* e, void* data) {
+  EMSCRIPTEN_WEBAUDIO_T ctx = VOIDP_2_WA(data);
+  if (emscripten_audio_context_state(ctx) != AUDIO_CONTEXT_STATE_RUNNING) {
+    printf("Resuming playback\n");
+    emscripten_resume_audio_context_sync(ctx);
+  }
+  printf("Toggling audio playback\n");
+  toggleTrack(beatID);
+    toggleTrack(bassID);
+  return false;
+}
+
+// Audio processor created, now register the audio callback
+void processorCreated(EMSCRIPTEN_WEBAUDIO_T context, bool success, void* data) {
+  if (success) {
```

Comment:
How about 

```
if (!success) { 
  printf("Audio worklet node creation failed\n");
  return;
}
```

Reply:
i.e. early return pattern

Reply:
I'm not a fan of an early return for simple cases, but I'm also not fussy. These are changed.